### PR TITLE
[New Rule] Microsoft 365 Teams Guest Access Enabled

### LINKED ISSUE #557: [New Rule] O365 Teams Guest Access Enabled
## Description
Identifies when Guest Access is enabled in Microsoft Teams. Guest access in Teams lets people outside your organization access teams and channels. An adversary may enable Guest Access to maintain persistence in an environment.

## Required Info

- **Eventing Sources:**
`filebeat-*`

- **Platforms**
`o365`

- **Target ECS Version:** `1.0.6`
- **New fields required in ECS for this?**
- **Related issues or PRs**

## Optional Info
- **References:**
https://docs.microsoft.com/en-us/powershell/module/skype/get-csteamsclientconfiguration?view=skype-ps

## Example Logic
```
event.dataset:o365.audit and event.provider:(SkypeForBusiness or MicrosoftTeams) and event.category:web and event.action:"Set-CsTeamsClientConfiguration" and o365.audit.Parameters.AllowGuestUser:True and event.outcome:success
```
## Screenshot 

image

### PR DESCRIPTION
**Pull request by bm11100**
## Issues
resolves #557 

## Summary
Identifies when guest access is enabled in Microsoft Teams. Guest access in Teams allows people outside the organization to access teams and channels. An adversary may enable guest access to maintain persistence in an environment.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by bm11100** [comment]
when #570 merges, the `non-ecs-schema` modification may no longer be needed. will confirm.

**Review by Samirbous — APPROVED** [review]

File: rules/o365/persistence_o365_teams_guest_access_enabled.toml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+[metadata]
+creation_date = "2020/11/20"
+ecs_version = ["1.6.0"]
```
Comment (dstepanic17) [inline rules/o365/persistence_o365_teams_guest_access_enabled.toml:3]:
```suggestion
```

**Review by dstepanic17 — APPROVED** [review]
LGTM, just left comment on `ecs_version`

**Comment by bm11100** [comment]
I still needed the `non-ecs-schema` even after #570 merged @brokensound77

**Comment by bm11100** [comment]
I still needed the `non-ecs-schema` even after #570 merged @brokensound77

**Review by devonakerr — APPROVED** [review]

**Pull request merged**